src/test/java/com/google/sps/login/LoginTest.java

```diff
@@ -0,0 +1,96 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.login;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+
+@RunWith(JUnit4.class)
+public final class LoginTest {
+
+    private LocalServiceTestHelper helper;
+	private Login login;
+
+    @Before
+    public void setUp() {
+        // instantiate Login class
+        login = new Login();
+    }
+
+    @Test
+    public void testIsLoggedIn() {
+        // set up testing environment where user is logged in
+        setUpHelper(true);
+
+        UserService userService = UserServiceFactory.getUserService();
+
+        // create expected authentication class that's logged in
+        String expectedLogoutURL = userService.createLogoutURL("/");
+        Authentication authExpected = new Authentication(true, expectedLogoutURL);
+
+        // get actual return from Login.java
+        Authentication authActual = login.getAuthentication(userService);
+
+        // check values of expected and actual to make sure they align
+        Assert.assertEquals(authExpected.getLoggedIn(), authActual.getLoggedIn());
+        Assert.assertEquals(authExpected.getRedirectLink(), authActual.getRedirectLink());
+
+        // tear down testing environment
+        helper.tearDown();
+    }
+
+    @Test
+    public void testIsLoggedOut() {
+        // set up testing environment where user is logged out
+        setUpHelper(false);
+
+        UserService userService = UserServiceFactory.getUserService();
+
+        // create expected authentication class that's logged in
+        String expectedLogoutURL = userService.createLoginURL("/");
+        Authentication authExpected = new Authentication(false, expectedLogoutURL);
+
+        // get actual return from Login.java
+        Authentication authActual = login.getAuthentication(userService);
+
+        // check values of expected and actual to make sure they align
+        Assert.assertEquals(authExpected.getLoggedIn(), authActual.getLoggedIn());
+        Assert.assertEquals(authExpected.getRedirectLink(), authActual.getRedirectLink());
+
+        // tear down testing environment
+        helper.tearDown();
+    }
+
+    // set up helper function depending on login status
+    public void setUpHelper(boolean loggedIn) {
+        if (loggedIn) {
```

Comment:
Lines 88 and 91 are exactly same (correct me if my eyes are tricking me).
Do you still need the if conditional?

Reply:
Ah right, no need for a conditional